reconcile/cli.py

```diff
@@ -376,6 +378,8 @@ def include_trigger_trace(function):
 
 
 def run_integration(func_container, ctx, *args, **kwargs):
+    faulthandler.enable(all_threads=True)
```

Comment:
@OliviaHY, this is nice! A small question here.

Would we need to ensure that the file descriptor the faulthanlder is using is stable?

My Python knowledge might be a bit dated. However, I remember it requiring the underlying file descriptor to be open for the lifespan of the faulthanlder (before it is disabled). Thus, if something does something silly to a standard error output, for example, we might have an issue.

For example, if something calls **dup()**/**dup2()** on the file descriptor the faulthanlder is using, it would stop working correctly.

Reply:
That's a great question @kwilczynski I did come across that in their [doc](https://docs.python.org/dev/library/faulthandler.html#issue-with-file-descriptors). 
It is an unlikely situation and when it does happen, all entries writing to stderr will be writing to a different file and not just faulthandler output, which is a much bigger problem for logging and monitoring.

Reply:
@OliviaHY, that would be not great, indeed. :smile: 

I am thinking more about the `sys.stderr`, which on some platforms (or under some circumstances) can be None (see: https://github.com/pyinstaller/pyinstaller/issues/4481), and there is also this:

- https://github.com/microsoft/vscode-python/issues/18346

This is something I stumbled upon once when I was using Visual Studio Code to debug a problem.

I was thinking more along the lines of ensuring that we have a stable file descriptor number for the standard error output, so using `sys.__stderr__`, for example:

```python
  def register_faulthandler(fileobj=sys.__stderr__):
      if fileobj is None:
          return
      try:
          import faulthandler
          faulthandler.disable()
          if (hasattr(faulthandler, 'register') and hasattr(signal, 'SIGUSR1')):
              faulthandler.register(signal.SIGUSR1, file=fileobj, all_threads=True)
      except (ImportError, RuntimeError, AttributeError):
          print('Could not enable faulthandler!')
```

The above example only registers the **SIGUSR1** signal handling. Also, I apologise for my sub-par Python! :bow: